src/amber/router/route_set.cr

```diff
@@ -147,18 +147,18 @@ module Amber::Router
     protected def reverse_select_routes(path : Array(String)) : Array(GlobMatch(T))
       matches = [] of GlobMatch(T)
 
-      @segments.each do |segment|
-        case segment
+      @segments.each do |s|
```

Comment:
I'd revert all changes within this file.

Reply:
Why?

Reply:
They are less expressive - `s` vs `segment`.

Reply:
originally was due to a shadowed variable ameba error, but must have went away after the merge because it works fine with `segment` now as well.